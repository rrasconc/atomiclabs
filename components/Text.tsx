import { Text as DefaultText, StyleSheet, TextProps } from 'react-native';

import { COLORS, FONT } from '../constants/theme';

export function Text({ style: customStyle, ...otherProps }: TextProps) {
  const style = [styles.text, customStyle];

  return <DefaultText style={style} {...otherProps} />;
}

const styles = StyleSheet.create({
  text: {
    color: COLORS.white,
    fontSize: FONT.medium,
  },
});
