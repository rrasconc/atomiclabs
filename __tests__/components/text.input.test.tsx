import { render, screen } from '@testing-library/react-native';

import { TextInput } from '../../components/Text.Input';

describe('<TextInput />', () => {
  let component: any = null;
  beforeEach(() => {
    component = render(<TextInput label="Name" error="There is an error!" />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows label', () => {
    expect(screen.getByText(/name/i)).toBeDefined();
  });

  it('shows error message', () => {
    expect(screen.getByText(/there is an error/i)).toBeDefined();
  });
});
