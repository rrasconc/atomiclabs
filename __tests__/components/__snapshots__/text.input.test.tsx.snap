// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders correctly 1`] = `
<View
  style={
    {
      "gap": 21.428571428571427,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 38.57142857142857,
        },
        {
          "fontWeight": "400",
        },
      ]
    }
  >
    Name
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "paddingHorizontal": 32.14285714285714,
        },
        {
          "borderColor": "#FF2626",
        },
      ]
    }
  >
    <TextInput
      style={
        {
          "flex": 1,
          "fontSize": 38.57142857142857,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "marginLeft": "auto",
          "padding": 21.428571428571427,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#707070",
              "fontSize": 38.57142857142857,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ï€£
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 38.57142857142857,
        },
        {
          "color": "#FF2626",
          "fontSize": 30,
          "fontWeight": "700",
          "marginTop": -10.714285714285714,
        },
      ]
    }
  >
    There is an error!
  </Text>
</View>
`;
